lockup/src/internal.rs

```diff
@@ -30,12 +30,19 @@ impl LockupContract {
         }
     }
 
-    pub fn assert_vesting(&self) {
-        if let VestingInformation::Vesting(_) = &self.vesting_information {
-            // OK
-        } else {
-            env::panic(b"There is no vesting in progress");
-        }
+    pub fn assert_vesting(&self, vesting_schedule: &VestingSchedule, salt: &[u8]) {
+        match &self.vesting_information {
+            VestingInformation::VestingHash(hash) => assert_eq!(
+                &hash_vesting_schedule(vesting_schedule, salt),
+                &hash.0,
+                "Presented vesting doesn't match or vesting was terminated"
+            ),
+            VestingInformation::VestingSchedule(vs) => assert_eq!(
```

Comment:
It seems like we support a scenario when vesting schedule is saved directly in the contract but then foundation still needs to provide it to call contract methods. It is just a weird UX. We could keep it to make things simple, but we should explain it in the comment, otherwise people would think there is a deeper meaning to this.

Reply:
Yep. Good point. Will update `terminate_vesting` method to take optional `VestingScheduleWithSalt`.